linkis-computation-governance/linkis-entrance/src/main/scala/org/apache/linkis/entrance/interceptor/impl/TenantData.scala

```diff
@@ -0,0 +1,100 @@
+import org.apache.linkis.governance.common.protocol.conf.{TenantRequest, TenantResponse}
+import org.apache.linkis.manager.label.builder.factory.LabelBuilderFactoryContext
+import org.apache.linkis.manager.label.constant.LabelKeyConstant
+import org.apache.linkis.manager.label.entity.TenantLabel
+import org.apache.linkis.manager.label.utils.LabelUtil
+import org.apache.linkis.rpc.Sender
+
+import org.apache.commons.lang3.StringUtils
+
+import java.{lang, util}
+import java.util.concurrent.TimeUnit
+
+import com.google.common.cache.{CacheBuilder, CacheLoader, LoadingCache}
+
+object TenantData extends Logging {
+
+  private val configCache: LoadingCache[String, String] = CacheBuilder
+    .newBuilder()
+    .maximumSize(1000)
+    .expireAfterAccess(1, TimeUnit.HOURS)
+    .refreshAfterWrite(EntranceConfiguration.USER_PARALLEL_REFLESH_TIME.getValue, TimeUnit.MINUTES)
+    .build(new CacheLoader[String, String]() {
+
+      override def load(userCreator: String): String = {
+        val sender: Sender = Sender
+          .getSender(Configuration.CLOUD_CONSOLE_CONFIGURATION_SPRING_APPLICATION_NAME.getValue)
+        val user = userCreator.split("-")(0)
+        val creator = userCreator.split("-")(1)
+        logger.info(s"load tentant data user $user creator $creator data")
+        sender.ask(TenantRequest(user, creator)) match {
+          case tenantResponse: TenantResponse => tenantResponse.tenant
+          case _ =>
+            logger.warn(s"TenantCache user $user creator $creator data loading failed")
+            ""
+        }
+      }
+
+    })
+
+  def checkTenantLabel(jobRequest: JobRequest, logAppender: lang.StringBuilder): JobRequest = {
+    jobRequest match {
+      case requestPersistTask: JobRequest =>
+        logger.info("start to checkTenantLabel")
+        var labels = requestPersistTask.getLabels
+        // 判断tenant 是否存在。存在则放行，不存在则进行回填
+        logger.info("check lalabels contains tenant :{} ", labels)
+        if (!labels.contains(LabelKeyConstant.TENANT_KEY)) {
+          // 获取user信息
+          val userName = jobRequest.getSubmitUser
+          // 未获取到用户信息报错
+          logger.info("userName {} ", userName)
+          if (StringUtils.isEmpty(userName)) {
+            throw TenantCheckException(
+              EntranceErrorCode.USER_NULL_EXCEPTION.getErrCode,
+              EntranceErrorCode.USER_NULL_EXCEPTION.getDesc
+            )
+          }
+          // 通过user-creator  获取缓存中的 tenant
+          val tenant = configCache.get(LabelUtil.getUserCreatorLabel(labels).getStringValue)
+          logger.info("get cache tenant {} ", tenant)
+          // 缓存获取数据不为空则添加进去
+          if (StringUtils.isNotBlank(tenant)) {
+            val tenantLabel = LabelBuilderFactoryContext.getLabelBuilderFactory
+              .createLabel[TenantLabel](LabelKeyConstant.TENANT_KEY)
+            tenantLabel.setTenant(tenant)
+            labels.add(tenantLabel)
+            logger.info("end to checkTenantLabel labels :{} ", labels)
```

Comment:
remove log update to LogUtils.generateInfo(""Your task should be to set tenant label$tenant") + "\n"

Reply:
done